File: Java/app/src/main/java/com/example/randomizermenumakanan/View/MainFragment.java
package com.example.randomizermenumakanan.View;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.example.randomizermenumakanan.Presenter.MainPresenter;
import com.example.randomizermenumakanan.databinding.FragmentMainBinding;


public class MainFragment extends Fragment implements View.OnClickListener {
    private FragmentListener listener;
    private FragmentMainBinding binding;
    private MainPresenter mainPresenter;

    public MainFragment(MainPresenter mainPresenter) {
        this.mainPresenter = mainPresenter;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        this.binding = FragmentMainBinding.inflate(inflater, container, false);
        this.binding.btnCari.setOnClickListener(this);
        return this.binding.getRoot();
    }

    @Override
    public void onClick(View v) {
        if (v == this.binding.btnCari) {
            this.listener.changePage(2);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof FragmentListener) {
            this.listener = (FragmentListener) context;
        } else {
            throw new ClassCastException(context.toString() + " must implement FragmentListener");
        }
    }
}

File: Kotlin/app/src/main/java/com/example/randomizermenumakanan/View/MainFragment.kt
package com.example.randomizermenumakanan.View

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import androidx.fragment.app.Fragment
import com.example.randomizermenumakanan.Model.IsiMenu
import com.example.randomizermenumakanan.R
import java.util.Random

class MainFragment : Fragment(), View.OnClickListener {
    private lateinit var etMain: EditText
    private lateinit var btnCari: Button
    private lateinit var itemList: List<IsiMenu>
    private lateinit var listener: FragmentListener

    private var rand: Random = Random()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_main, container, false)
        this.btnCari = view.findViewById(R.id.btnCari)
        this.btnCari.setOnClickListener(this)
        return view
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is FragmentListener) {
            listener = context
        } else {
            throw ClassCastException("$context must implement FragmentListener")
        }
    }

    override fun onClick(v: View) {
        if(v == this.btnCari) {
//            val choice = this.rand.nextInt(itemList.size)
//            Log.d("random", choice.toString())
            listener.changePage(4)


        }
    }
}